src/jit/lir.cpp

```diff
@@ -578,6 +759,62 @@ bool LIR::Range::TryGetUse(GenTree* node, Use* use)
     return false;
 }
 
+//------------------------------------------------------------------------
+// LIR::Range::GetTreeRange: Computes the subrange that includes all nodes
+//                           in the dataflow tree rooted at a particular
+//                           node.
+//
+// Arguments:
+//    root     - The root of the dataflow tree.
+//    isClosed - An output parameter that is set to true if the returned
+//               range contains only nodes in the dataflow tree and false
+//               otherwise.
+//
+// Returns:
+//    The computed subrange.
+LIR::Range LIR::Range::GetTreeRange(GenTree* root, bool* isClosed) const
+{
+    assert(root != nullptr);
+    assert(isClosed != nullptr);
+
+    // Mark the root of the tree
```

Comment:
It would be nice to have a short overall description of the algorithm here. (this could be TBD)